public/documentation/ios.md

```diff
@@ -0,0 +1,176 @@
+
+3. Create The Echo Extension
+
+  * Create a framework target called `EchoExtension` in the `Echo` project.
+
+    In `File` -> `New...` -> `Target...`, choose `Framework & Library` -> `Cocoa Touch Framework`, with name `EchoExtension` and language type `Swift` for convenience.
+
+  * Link `XWalkView` into `EchoExtension` target.
+
+    Select `EchoExtension` target in `Echo` project, in `General` -> `Linked Frameworks and Libraries`, add `XWalkView.framework`.
+
+  * Create the `EchoExtension` extension class.
+
+    Select `EchoExtension` group first, in `File` -> `New...` -> `File...`, choose `iOS` -> `Source` -> `Cocoa Touch Class`, with name `EchoExtension`, subclassing from `XWalkExtension`, and use `Swift` language type.
+
+  * Add the contents into `EchoExtension.swift` as follows:
+
+  ```swift
+  import Foundation
+  import XWalkView
+
+  class EchoExtension : XWalkExtension {
+    func jsfunc_echo(cid: UInt32, message: String, callback: UInt32) -> Bool {
+          invokeCallback(callback, key: nil, arguments: ["Echo from native: " + message])
+          return true
+      }
+  }
+  ```
+
+  * Then we need to add the extension description section in the `Info.plist` of the `EchoExtension` target, to make the framework users aware that what kind of extensions are provided by this framework, and what are the extension names in both native and JavaScript world.
+
+    * Create `XWalkExtensions` section in the `Info.plist` with `Dictionary` type, then add an entry with `xwalk.example.echo` as key and `EchoExtension` as value.
+      ![Info.plist](https://cloud.githubusercontent.com/assets/700736/7226047/58728d94-e777-11e4-9fd4-8d23a24d981f.png)
+
+    This indicates that the extension `EchoExtension` will be exported with the object name: `xwalk.example.echo` in JavaScript.
+
+4. Bundle The Extension With The Application
+
+  * Modify the `manifest.plist` to add the extension configuration to the `Echo` application.
+
+    * Add `xwalk_extensions` section with Array type which describes the extensions that should be loaded within the `Echo` application;
+
+    * And add `xwalk.example.echo` with String type as an entry of `xwalk_extensions`, which indicates that the extension `xwalk.example.echo` should be loaded into the JavaScript context of `Echo` project.
+      ![manifest2](https://cloud.githubusercontent.com/assets/700736/7226213/3ef59a9e-e779-11e4-822f-1ef6775723ad.png)
+
+  * Then you can build and run the application to test. If everything goes well, you can see an extra line: `Echo from native: Hello World!` displayed on the screen.
+
+For further information please read the [Getting Started Guide](https://github.com/crosswalk-project/crosswalk-ios/wiki/Getting-Started-With-Crosswalk-for-iOS), and other articles on the [Wiki](https://github.com/crosswalk-project/crosswalk-ios/wiki).
```

Comment:
Are we going to have a "Getting started guide" on the wiki?  If so, why not in the formal docs.   If we have this in the documentation then the guide needs to be done before release.